RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderEDProducer.cc

```diff
@@ -0,0 +1,130 @@
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderEDProducer.h"
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithm.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+VectorHitBuilderEDProducer::VectorHitBuilderEDProducer(edm::ParameterSet const& conf)
+    : offlinestubsTag_(conf.getParameter<std::string>("offlinestubs")),
+      maxOfflinestubs_(conf.getParameter<int>("maxVectorHits")),
+      algoTag_(conf.getParameter<std::string>("Algorithm")),
+      //_clusterProducer(conf.getParameter<edm::InputTag>("Clusters")),
+      readytobuild_(false) {
+  clusterProducer_ =
+      consumes<edmNew::DetSetVector<Phase2TrackerCluster1D>>(edm::InputTag(conf.getParameter<std::string>("Clusters")));
+
+  produces<edmNew::DetSetVector<Phase2TrackerCluster1D>>("ClustersAccepted");
+  produces<edmNew::DetSetVector<Phase2TrackerCluster1D>>("ClustersRejected");
+  produces<VectorHitCollectionNew>(offlinestubsTag_ + "Accepted");
+  produces<VectorHitCollectionNew>(offlinestubsTag_ + "Rejected");
+  setupAlgorithm(conf);
+}
+
+VectorHitBuilderEDProducer::~VectorHitBuilderEDProducer() { delete stubsBuilder_; }
+
+void VectorHitBuilderEDProducer::produce(edm::Event& event, const edm::EventSetup& es) {
+  LogDebug("VectorHitBuilderEDProducer") << "VectorHitBuilderEDProducer::produce() begin";
+
+  // get input clusters data
+  edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clustersHandle;
+  event.getByToken(clusterProducer_, clustersHandle);
+
+  // create the final output collection
+  std::unique_ptr<edmNew::DetSetVector<Phase2TrackerCluster1D>> outputClustersAccepted(
+      new edmNew::DetSetVector<Phase2TrackerCluster1D>);
+  std::unique_ptr<edmNew::DetSetVector<Phase2TrackerCluster1D>> outputClustersRejected(
+      new edmNew::DetSetVector<Phase2TrackerCluster1D>);
+  std::unique_ptr<VectorHitCollectionNew> outputVHAccepted(new VectorHitCollectionNew());
+  std::unique_ptr<VectorHitCollectionNew> outputVHRejected(new VectorHitCollectionNew());
+
+  if (readytobuild_)
+    stubsBuilder_->initialize(es);
+  else
+    edm::LogError("VectorHitBuilderEDProducer") << "Impossible initialization of builder!!";
+
+  // check on the input clusters
+  stubsBuilder_->printClusters(*clustersHandle);
+
+  // running the stub building algorithm
+  //ERICA::output should be moved in the different algo classes?
+  run(clustersHandle, *outputClustersAccepted, *outputClustersRejected, *outputVHAccepted, *outputVHRejected);
+
+  unsigned int numberOfVectorHits = 0;
+  edmNew::DetSetVector<VectorHit>::const_iterator DSViter;
+  for (DSViter = (*outputVHAccepted).begin(); DSViter != (*outputVHAccepted).end(); DSViter++) {
```

Comment:
might be preferable to put this loop, which seems only to exist for debugging, behind preprocessor flags (`EDM_ML_DEBUG`)

Reply:
Done